gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -584,6 +587,36 @@ private static void setupGenericConjureProjects(
         });
     }
 
+    /**
+     * Locates projects either as child projects or as peer projects whose names match the patterns given by
+     * the GENERIC_GENERATOR_LANGUAGE_NAMES_PROPERTY property.
+     */
+    private static Map<String, Project> findGenericDerivedProjects(Project project) {
+        Map<String, Project> genericSubProjects = Collections.emptyMap();
+        String projectName = project.getName();
+
+        boolean useFlatProjectStructure = project.hasProperty(USE_FLAT_PROJECT_STRUCTURE_PROPERTY);
+        if (!useFlatProjectStructure) {
+            genericSubProjects = Maps.filterKeys(project.getChildProjects(), childProjectName -> {
+                return childProjectName.startsWith(projectName)
+                        && !FIRST_CLASS_GENERATOR_PROJECT_NAMES.contains(
+                                extractSubprojectLanguage(projectName, childProjectName));
+            });
+        } else {
+            if (project.hasProperty(GENERIC_GENERATOR_LANGUAGE_NAMES_PROPERTY)) {
```

Comment:
this is necessary for our use case because with the flat project structure, we can't actually infer the difference between a top-level Conjure API project, where this plugin is applied, and a pseudo-subproject that receives the output of a custom generator.

As an example, we have both a `foo-api` and a `foo-api-internal` project, both of which define Conjure APIs, and both of which use a custom generator (so we also have `foo-api-custom` and `foo-api-internal-custom`). Without the explicit hint here, `foo-api-internal` could be interpreted as a pseudo-subproject of `foo-api`, for the "internal" custom generator type.

Reply:
Is there any way we can invert this logic to check the `conjureGenerators` dependencies first to compute a list of possible conjure module names? That way we don't have to worry about suffix ambiguity.

Reply:
My understanding is that we can't do that because the code here: https://github.com/palantir/gradle-conjure/blob/develop/gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java#L516-L585 needs to resolve the `conjureGenerators` configuration in order to get a list of names, which has to happen after the projects are evaluated. But, that code also needs to setup tasks on the project that will receive the generator output.

@AlexLandau had a separate idea of separating this code out into a different gradle plugin that is actually applied on the receiver project (instead of the root API project), and placing the `conjureGenerators` configuration there. I think that would solve the problem for our flat project structure use case, with the tradeoff being that it's more boilerplate to have to manually add.